src/codeSigning/packAndSign.ts

```diff
@@ -454,4 +460,54 @@ export const api = {
       }
     }
   },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  getAgentForUri(url: string): false | Agents {
```

Comment:
I'll add some comments to the function.

Its job is to determine if a proxy agent is necessary and then pick the right one based both protocol of the request url and from which env var the proxy URLs are passed in.

Also checks the domain name against `no_proxy` env var. No proxy needed if there is a hit.

Reply:
Why not import a library that's able to handle proxies own its own?

Reply:
@mdonnalley I found one `proxy-agent` that handled all this. When I tried to compile with it I was getting errors about a library named `agent-base` and one of its imports not being exported.

Reply:
@mdonnalley @amphro @mshanemc I was able to get proxy-agent to work, but I had to add `"allowSyntheticDefaultImports": true` to tsconfig.